action.yml

```diff
@@ -0,0 +1,62 @@
+name: 'Create Mender deployment'
+description: 'Create a deployment on a hosted Mender backend instance'
+inputs:
+  mender_pat:
+    description: 'personal access token for the hosted Mender backend account to be used'
+    required: true
+  mender_deployment_name:
+    description: 'deployment name to be created'
+    required: true
+  mender_release_name:
+    description: 'Mender artifacts release name'
+    required: true
+  mender_devices_group:
+    description: 'the name of the devices group to which the release will be deployed'
+    required: false
+  mender_devices_list:
+    description: 'the list of device ids to which the release will be deployed'
+    required: false
+  mender_uri:
+    description: 'URI for the Mender server to be used'
+    required: false
+    default: 'https://hosted.mender.io'
+runs:
+  using: "composite"
+  steps:
+    - id: input_validation
+      shell: bash
+      run: |
+        if [[ -z "${{ inputs.mender_devices_group }}" && -z "${{ inputs.mender_devices_list }}" ]]; then
+          echo "ERROR: both mender_devices_group and mender_devices_list are empty, at least one is expected to be set"
+          exit 1
+        fi
```

Comment:
We actually do input validation there as well by checking if an artifact exists, it just does't have an error message.

The reason for this check is `required: false` for both of the inputs in the action definition. And the reason for that is that I want to make it more unified and support both use cases, so, have to do it manually here instead of delegating it to the framework.